packages/shared/ReactComponentStackFrame.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Source} from 'shared/ReactElementType';
+import type {LazyComponent} from 'react/src/ReactLazy';
+
+import {
+  REACT_SUSPENSE_TYPE,
+  REACT_SUSPENSE_LIST_TYPE,
+  REACT_FORWARD_REF_TYPE,
+  REACT_MEMO_TYPE,
+  REACT_BLOCK_TYPE,
+  REACT_LAZY_TYPE,
+} from 'shared/ReactSymbols';
+
+const BEFORE_SLASH_RE = /^(.*)[\\\/]/;
+
+function describeComponentFrame(
+  name: null | string,
+  source: void | null | Source,
+  ownerName: null | string,
+) {
+  let sourceInfo = '';
+  if (__DEV__ && source) {
+    const path = source.fileName;
+    let fileName = path.replace(BEFORE_SLASH_RE, '');
+    // In DEV, include code for a common special case:
+    // prefer "folder/index.js" instead of just "index.js".
+    if (/^index\./.test(fileName)) {
+      const match = path.match(BEFORE_SLASH_RE);
+      if (match) {
+        const pathBeforeSlash = match[1];
+        if (pathBeforeSlash) {
+          const folderName = pathBeforeSlash.replace(BEFORE_SLASH_RE, '');
+          fileName = folderName + '/' + fileName;
+        }
+      }
+    }
+    sourceInfo = ' (at ' + fileName + ':' + source.lineNumber + ')';
+  } else if (ownerName) {
+    sourceInfo = ' (created by ' + ownerName + ')';
+  }
+  return '\n    in ' + (name || 'Unknown') + sourceInfo;
+}
+
+export function describeBuiltInComponentFrame(
+  name: string,
+  source: void | null | Source,
+  ownerFn: void | null | Function,
+): string {
+  let ownerName = null;
+  if (__DEV__ && ownerFn) {
+    ownerName = ownerFn.displayName || ownerFn.name || null;
+  }
+  return describeComponentFrame(name, source, ownerName);
+}
+
+export function describeClassComponentFrame(
+  ctor: Function,
+  source: void | null | Source,
+  ownerFn: void | null | Function,
+): string {
+  return describeFunctionComponentFrame(ctor, source, ownerFn);
```

Comment:
This looks like a bug. `ownerFn` should not be passed directly. Why didn't Flow catch this?

Reply:
Oh. Whoops. I misread this as calling `describeComponentFrame`. My bad.